statsmodels/discrete/count_model.py

```diff
@@ -0,0 +1,730 @@
+                    self.exog_infl[zero_idx, i] / pmf[zero_idx]).sum()
+        
+        return hess_arr
+
+    def _hessian_probit(self, params):
+        pass
+
+    def hessian(self, params):
+        """
+        Generic Zero Inflated model Hessian matrix of the loglikelihood
+
+        Parameters
+        ----------
+        params : array-like
+            The parameters of the model
+
+        Returns
+        -------
+        hess : ndarray, (k_vars, k_vars)
+            The Hessian, second derivative of loglikelihood function,
+            evaluated at `params`
+
+        Notes
+        -----
+        """
+        hess_arr_main = self._hessian_main(params)
+        hess_arr_infl = self._hessian_inflate(params)
+
+        if hess_arr_main is None or hess_arr_infl is None:
+            return approx_hess(params, self.loglike)
+
+        dim = self.k_exog + self.k_inflate
+
+        hess_arr = np.zeros((dim, dim))
+
+        hess_arr[:self.k_inflate,:] = hess_arr_infl
+        hess_arr[self.k_inflate:,self.k_inflate:] = hess_arr_main
+
+        tri_idx = np.triu_indices(self.k_exog + self.k_inflate, k=1)
+        hess_arr[tri_idx] = hess_arr.T[tri_idx]
+
+        return hess_arr
+
+    def predict(self, params, exog=None, exog_infl=None, exposure=None,
+                offset=None, which='mean'):
+        """
+        Predict response variable of a count model given exogenous variables.
+
+        Parameters
+        ----------
+        params : array-like
+            The parameters of the model
+        exog : array, optional
+            A reference to the exogenous design.
+            If not assigned, will be used exog from fitting.
+        exog_infl : array, optional
+            A reference to the zero-inflated exogenous design.
+            If not assigned, will be used exog from fitting.
+        offset : array, optional
+            Offset is added to the linear prediction with coefficient equal to 1.
+        exposure : array, optional
+            Log(exposure) is added to the linear prediction with coefficient
+            equal to 1. If exposure is specified, then it will be logged by the method.
+            The user does not need to log it first.
+        which : string, optional
+            Define values that will be predicted.
+            'mean', 'mean-main', 'linear', 'mean-nonzero', 'prob-zero, 'prob', 'prob-main'
+            Default is 'mean'.
+
+        Notes
+        -----
+        """
+        if exog is None:
+            exog = self.exog
+
+        if exog_infl is None:
+            exog_infl = self.exog_infl
+
+        if exposure is None:
+            exposure = getattr(self, 'exposure', 0)
+        elif exposure != 0:
+            exposure = np.log(exposure)
+
+        if offset is None:
+            offset = 0
+
+        params_infl = params[:self.k_inflate]
+        params_main = params[self.k_inflate:]
+
+        prob_main = 1 - self.model_infl.predict(params_infl, exog_infl).mean()
```

Comment:
I think there should NOT be a `.mean()` here

The returned `which='prob-main'`  needs to have the same length or nobs as the exog.

Also, I think in the computation of the other `which`, it should use the observation or exog specific prob_main and not the average prob_main